base/services/w32time/W32Time.c

```diff
@@ -0,0 +1,370 @@
+                                  SE_SYSTEMTIME_NAME,
+                                  &priv.Privileges[0].Luid))
+        {
+            if (AdjustTokenPrivileges(hToken,
+                                      FALSE,
+                                      &priv,
+                                      sizeof(previouspriv),
+                                      &previouspriv,
+                                      &PrevSize) &&
+                GetLastError() == ERROR_SUCCESS)
+            {
+                /*
+                 * We successfully enabled it, we're permitted to change the time.
+                 * Check the second parameter for SystemTime and if TRUE set System Time.
+                 * Otherwise, if FALSE set the Local Time.
+                 * Call SetLocalTime twice to ensure correct results.
+                 */
+                if (SystemTime)
+                {
+                    Ret = SetSystemTime(lpSystemTime);
+                }
+                else
+                {
+                    Ret = SetLocalTime(lpSystemTime) &&
+                          SetLocalTime(lpSystemTime);
+                }
+
+                /*
+                 * For the sake of security, restore the previous status again
+                 */
+                if (previouspriv.PrivilegeCount > 0)
+                {
+                    AdjustTokenPrivileges(hToken,
+                                          FALSE,
+                                          &previouspriv,
+                                          0,
+                                          NULL,
+                                          0);
+                }
+            }
+        }
+        CloseHandle(hToken);
+    }
+
+    return Ret;
+}
+
+static DWORD
+GetIntervalSetting()
+{
+    HKEY hKey;
+    DWORD dwData;
+    DWORD dwSize = sizeof(DWORD);
+
+    dwData = 0;
+    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                      L"SYSTEM\\CurrentControlSet\\Services\\W32Time\\TimeProviders\\NtpClient",
+                      0,
+                      KEY_QUERY_VALUE,
+                      &hKey) == ERROR_SUCCESS)
+    {
+        RegQueryValueExW(hKey,
+                             L"SpecialPollInterval",
+                             NULL,
+                             NULL,
+                             (LPBYTE)&dwData,
+                             &dwSize);
+        RegCloseKey(hKey);
+    }
+    return dwData;
+}
+
+int set_time()
+{
+    ULONG ulTime;
+    LONG lRet;
+    HKEY hKey;
+    DWORD dwIndex = 0;
+    DWORD dwValSize;
+    DWORD dwNameSize;
+    WCHAR szValName[MAX_VALUE_NAME];
+    WCHAR szData[256];
+    WCHAR szDefault[256];
+
+
+    lRet = RegOpenKeyExW(HKEY_LOCAL_MACHINE,
+                        L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\DateTime\\Servers",
+                        0,
+                        KEY_QUERY_VALUE,
+                        &hKey);
+    if (lRet != ERROR_SUCCESS)
+        {
+        return 1;
+        }
```

Comment:
Broken indenting

Reply:
I will move the braces back 4 spaces to be under the "if".